examples/mugen/retrieval/s3d.py

```diff
@@ -0,0 +1,372 @@
+        super(S3D, self).__init__()
+        self.base = nn.Sequential(
+            SepConv3d(3, 64, kernel_size=7, stride=2, padding=3),
+            nn.MaxPool3d(kernel_size=(1, 3, 3), stride=(1, 2, 2), padding=(0, 1, 1)),
+            BasicConv3d(64, 64, kernel_size=1, stride=1),
+            SepConv3d(64, 192, kernel_size=3, stride=1, padding=1),
+            nn.MaxPool3d(kernel_size=(1, 3, 3), stride=(1, 2, 2), padding=(0, 1, 1)),
+            Mixed3b(),
+            Mixed3c(),
+            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1)),
+            Mixed4b(),
+            Mixed4c(),
+            Mixed4d(),
+            Mixed4e(),
+            Mixed4f(),
+            nn.MaxPool3d(kernel_size=(2, 2, 2), stride=(2, 2, 2), padding=(0, 0, 0)),
+            Mixed5b(),
+            Mixed5c(),
+        )
+        self.fc = nn.Sequential(
+            nn.Conv3d(1024, num_class, kernel_size=1, stride=1, bias=True),
+        )
+
+    def forward(self, x):
+        y = self.base(x)
+        y = F.avg_pool3d(y, (2, y.size(3), y.size(4)), stride=1)
+        y = self.fc(y)
+        y = y.view(y.size(0), y.size(1), y.size(2))
+        logits = torch.mean(y, 2)
+
+        return logits
+
+
+class BasicConv3d(nn.Module):
+    def __init__(self, in_planes, out_planes, kernel_size, stride, padding=0):
+        super(BasicConv3d, self).__init__()
+        self.conv = nn.Conv3d(
+            in_planes,
+            out_planes,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            bias=False,
+        )
+        self.bn = nn.BatchNorm3d(out_planes, eps=1e-3, momentum=0.001, affine=True)
+        self.relu = nn.ReLU()
+
+    def forward(self, x):
+        x = self.conv(x)
+        x = self.bn(x)
+        x = self.relu(x)
+        return x
+
+
+class SepConv3d(nn.Module):
+    def __init__(self, in_planes, out_planes, kernel_size, stride, padding=0):
+        super(SepConv3d, self).__init__()
+        self.conv_s = nn.Conv3d(
+            in_planes,
+            out_planes,
+            kernel_size=(1, kernel_size, kernel_size),
+            stride=(1, stride, stride),
+            padding=(0, padding, padding),
+            bias=False,
+        )
+        self.bn_s = nn.BatchNorm3d(out_planes, eps=1e-3, momentum=0.001, affine=True)
+        self.relu_s = nn.ReLU()
+
+        self.conv_t = nn.Conv3d(
+            out_planes,
+            out_planes,
+            kernel_size=(kernel_size, 1, 1),
+            stride=(stride, 1, 1),
+            padding=(padding, 0, 0),
+            bias=False,
+        )
+        self.bn_t = nn.BatchNorm3d(out_planes, eps=1e-3, momentum=0.001, affine=True)
+        self.relu_t = nn.ReLU()
+
+    def forward(self, x):
+        x = self.conv_s(x)
+        x = self.bn_s(x)
+        x = self.relu_s(x)
+
+        x = self.conv_t(x)
+        x = self.bn_t(x)
+        x = self.relu_t(x)
+        return x
+
+
+class Mixed3b(nn.Module):
```

Comment:
do you know the reason for the Mixed names? although I can't think of a clearer name for these...

Reply:
because they use a mix of 3D and 2D convolutions! [paper](https://arxiv.org/pdf/1712.04851.pdf) page 3

Reply:
I'm not sure about the 3b part though

Reply:
maybe you could rename to MixedConvs3b? your call though

Reply:
Possibly a noob suggestion here, but all these `Mixed*` classes have very similar structure. Why not create a class per branch (except maybe `branch0` since it's just a single op) that you can pass the primitive params (in/out dims, kernel_size, stride) to, then write a single class to handle the `torch.cat` portion of things? Then you don't have to write the same 4 sequentials a ton of times

Reply:
Lan and I discussed not generalizing the S3D architecture yet, since that process requires more time to understand the existing torchvision VideoResnet code and generalize S3D in a consistent way (reusing torchvision classes, generalizing torchvision classes, following similar submodule structure). I like your suggestions but for now we didn’t want to generalize this S3D code if it has to be changed later anyway to a possibly different structure.

Reply:
I am not necessarily thinking about how to generalize the S3D architecture as a whole, more about how to reduce the copypasta in all the `Mixed*` classes. Seems they could all come from a common class with a different set of params passed, no?

Reply:
Sure I'll try that out in a new commit